src/v2/view-builder/internals/BaseOktaVerifyChallengeView.js

```diff
@@ -14,7 +14,13 @@ const request = (opts) => {
     method: 'GET',
     contentType: 'application/json',
   }, opts);
-  return $.ajax(ajaxOptions);
+  return new Promise((resolve, reject) => {
```

Comment:
Explain why/how this helps

Reply:
@shuowu-okta did you have a specific behavior-change related concern w.r.t. Promisifying this?

Reply:
In current code Promises are mixed with jQuery's Deferred  (line 159 `let probeChain = Promise.resolve();` uses Promise, ` $.ajax` returns Deferred and `done/fail` APIs are used)

Tests expect that requests are performed in chain: `probe port 2000 -> challenge 2000 (if probe OK) -> probe 6511 -> challenge 6511 -> probe 6512 -> challenge 6512 -> probe 6513 -> challenge 6513`
With current code it's not always true, eg. `probe 6511` can run in parallel with `challenge 2000`
So eg. this check
https://github.com/okta/okta-signin-widget/blob/054cb76b313d5333dd4028833a956028ac4289de/test/testcafe/spec/DeviceChallengePollView_spec.js#L435
can fail

Reply:
Changing the base xhr approach may cause unexpected issue for legacy browsers. As the goal is to fix flaky bacon test, I propose to find another way that fix against test directly, for example, migrate test from e2e to unit level (testcafe -> jest).

Reply:
See latest commit that uses `$.Deferred` instead of `Promise` and `then+catch` API instead of `done+fail` for proper chaining.
The recommended API method to use chaining is `then`: https://api.jquery.com/deferred.then/